quadsim/robots/motor_model.py

```diff
@@ -32,71 +32,85 @@ class MotorModel:
       torque is a sum of PD torque and additional torque.
     """
 
-    def __init__(self, config) -> None:
+    def __init__(
+        self,
+        motor_control_mode: MotorControlMode,
+        min_position: float,
+        max_position: float,
+        min_velocity: float,
+        max_velocity: float,
+        min_torque: float,
+        max_torque: float,
+        kp: float,
+        kd: float,
+    ) -> None:
         """Initializes the class."""
-        self.config = config
-        self._motor_control_mode = self.config.motor_control_mode
-        self._num_motors = self.config.num_motors
-        self._kp = self.config.kps
-        self._kd = self.config.kds
-        self._strength_ratios = np.ones(self._num_motors)
+        self._motor_control_mode = motor_control_mode
+        self._kp = kp
+        self._kd = kd
+        self._strength_ratio = 1
         # Thresholds
-        self._max_torque = self.config.max_torque
-        self._min_torque = self.config.min_torque
-        self._max_velocity = self.config.max_velocity
-        self._min_velocity = self.config.min_velocity
-        self._max_position = self.config.max_position
-        self._min_position = self.config.min_position
-
-    def set_motor_gains(
-        self, kp: Union[float, Sequence[float]], kd: Union[float, Sequence[float]]
-    ) -> None:
-        self._kp = np.ones(self._num_motors) * kp
-        self._kd = np.ones(self._num_motors) * kd
+        self._max_torque = max_torque
+        self._min_torque = min_torque
+        self._max_velocity = max_velocity
+        self._min_velocity = min_velocity
+        self._max_position = max_position
+        self._min_position = min_position
 
-    def set_strength_ratios(
-        self, strength_ratios: Union[float, Sequence[float]]
-    ) -> None:
-        self._strength_ratios = np.ones(self._num_motors) * strength_ratios
+    def set_motor_gain(self, kp: float, kd: float) -> None:
+        self._kp = kp
+        self._kd = kd
 
-    def _clip_torques(
-        self, desired_torque: Sequence[float], current_motor_velocity: Sequence[float]
-    ):
+    def set_strength_ratio(self, strength_ratio: float) -> None:
```

Comment:
I am willing to be convinced that we retain these 'setters', but I just want to talk a bit about it before hand =]. I don't find myself writing setters much in python code.

I will say that if an attribute is set a lot, using a function that wraps it can add a bit of overhead. Not sure if it's really significant in this case though.